### PR TITLE
[VA-API] Fix filter order

### PR DESCRIPTION
**Pull request by lbeltrame**
ffmpeg is very picky about the filters to be used when using VA-API,   because most of them are incompatible. This is particularly evident when   burning-in subtitles.
 
This commit:   
    - adds new video filters to the VA-API ffmpeg command;
    - ensures `hwmap` is also last in the line.

 Tests done:    
    - Played a video file with subtitles
    - Played a video file without subtitles
    
Note that far more testing is required to make sure all corner cases are doing OK.
    
Fixes issue #642.

This is a rebased branch of my previous PR (#694) on the new master branch. I think I already answered the concerns raised there:

- Options for VA-API are there before ffmpeg 4.0
- They're only applied for x264, other cases aren't even covered.

### CODE REVIEW

File: MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs [hunk 1]
```diff
@@ -1743,6 +1743,19 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
 
                 filters.Add(subParam);
 
+		// Ensure proper filters are passed to ffmpeg in case of hardware acceleration via VA-API
+		// Reference: https://trac.ffmpeg.org/wiki/Hardware/VAAPI
+		if (string.Equals(outputVideoCodec, "h264_vaapi", StringComparison.OrdinalIgnoreCase))
```
Comment (marius-luca-87) [inline MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs]: Merge issue.

Comment (lbeltrame) [reply MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs]: Fixed.

**Comment by lbeltrame** [comment]
Ok, I already found some possible not-so-corner case (I tested `ffmpeg` directly because it's easier): the `hwmap` and co bits need to go only if there are subtitles there, or ffmpeg will error out with `failed to map frame` (at least in the videos I tested it with). 

As I barely understand the code, what would be the best way to check if there are subtitles?

**Comment by lbeltrame** [comment]
Yes, I'm still running this on my instance at the moment.

**Comment by MatMaul** [comment]
@lbeltrame  Was a bit slow to review but thx for that :)

**Comment by bobberb** [comment]
Thanks for this.
